src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -232,6 +232,12 @@ internal Binder GetSpeculativeBinder(int position, ExpressionSyntax expression,
 
             binder = new WithNullableContextBinder(SyntaxTree, position, binder);
 
+            // PROTOTYPE(semi-auto-props): ContainingMember() to support local functions and lambdas.
+            if (binder.ContainingMemberOrLambda is SourcePropertyAccessorSymbol { Property.IsIndexer: false } accessor)
```

Comment:
>Property.IsIndexer: false [](http://example.com/codeflow?start=82&length=25)

It looks like other similar places are missing this condition (below for example). How necessary is it? It feels like it would be good to have some consistency. Perhaps we could extract a reusable helper for the entire ```if``` condition and use it everywhere where we are making a decision about adding a specialized Field binder.

Reply:
I was expecting this refactoring to be the next PR.

Reply:
The check isn't necessary, I think. While binding, we only use field keyword binder if the property is not an indexer. This just saves creating an object that we don't use. Will make it consistent.